x-pack/plugins/snapshot_restore/public/application/components/policy_form/steps/step_logistics.tsx

```diff
@@ -257,7 +273,7 @@ export const PolicyStepLogistics: React.FunctionComponent<StepProps> = ({
       setShowRepositoryNotFoundWarning(!doesRepositoryExist);
     }
 
-    return (
+    return disableToolTip(
```

Comment:
Why don't we reuse the new `DisableToolTip` component here instead of this function?

Reply:
It's because the tooltip message is different for the managed repository and the managed policy. But if we want to handle both cases (managed repository and managed policy) within the same component, I guess we can refactor the `DisableTooltip` component to something like:

**x-pack/plugins/snapshot_restore/public/application/components/disable_tooltip.tsx**

```js
export const DisableToolTip: React.FunctionComponent<Props> = ({
  isManagedRepository,
  isManagedPolicy,
  component,
}) => {
  let tooltipContent = '';

  if (isManagedRepository) {
    tooltipContent = i18n.translate('xpack.snapshotRestore.repositoryForm.disableToolTipContent', {
      defaultMessage: 'This field is disabled because you are editing a managed repository.',
    });
  } else if (isManagedPolicy) {
    tooltipContent = i18n.translate('xpack.snapshotRestore.policyForm.disableToolTipContent', {
      defaultMessage: 'This field is disabled because you are editing a managed policy.',
    });
  }

  return isManagedRepository || isManagedPolicy ? (
    <EuiToolTip content={tooltipContent} display="block">
      {component}
    </EuiToolTip>
  ) : (
    component
  );
};
```

Let me know what you think 😃

Reply:
Ah I see... I would suggest that you pass the message as a prop to the component and rename the `isManagedRepository` prop to just `isManaged`. Something like this:
```
export const DisableToolTip: React.FunctionComponent<Props> = ({
  isManaged,
  tooltipMessage,
  component,
}) => {
  return isManaged ? (
    <EuiToolTip
      content={tooltipMessage}
      display="block"
    >
      {component}
    </EuiToolTip>
  ) : (
    component
  );
};
```

Also, could you please add documentation above this component to explain what we use it for?

Reply:
and add a `const tooltipMessage` to define the tooltip message for each of the files, correct? (i.e. `azure_settings.tsx`, `gcs_settings.tsx`, `s3_settings.tsx` and `step_logistics.tsx`)

Reply:
For example:
```
const tooltipMessage = i18n.translate('xpack.snapshotRestore.repositoryForm.disableToolTip', {
    defaultMessage: 'This field is disabled because you are editing a managed repository.',
  });
```

Reply:
Yeah, I think it would be nice to add two constants (one for the repo message and one for the policy message) in `x-pack/plugins/snapshot_restore/public/application/constants/index.ts`. You could name them something more descriptive like `managed<Policy/Repository>FieldTooltipMessage`.